validation/edge/edge_validation.py

```diff
@@ -11,154 +11,122 @@
-path_to_file = pjoin("output", filename)
-#gdf_study_area = pd.merge(gdf_study_area, gdf_study_area.bounds, left_index=True, right_index=True)
-
-if os.path.isfile(path_to_file):
-    print ("File exist")
-else:
-    gdf_study_area.to_file(path_to_file)
-
-
-#Calculating the total length of osm network
-totallength = 0
-count_osm = 0
-osmlength = osm_data.length
-for i in osmlength:
-    count_osm += 1
-    totallength += i
-print(totallength, count_osm)
-
-
-#Calculate total length of official network
-totallength2 = 0
-count_of = 0
-oflength = official_data.length
-for i in oflength:
-    count_of += 1
-    totallength2 += i
-print(totallength2, count_of)
-
-#calculate areal intersection with various buffering
-buff = 5
-percent_list = []
-percent_dict = {}
-
-while buff<20:
-
-
-    official_buffer = official_data
-    official_buffer = official_buffer.buffer(buff)
-
-    osm_buffer = osm_data
-    osm_buffer = osm_buffer.buffer(buff)
-
-
-    osm_total = 0
-    osm_areas = osm_buffer.area
-    for i in osm_areas:
-        osm_total += i
-   # print(osm_areas[0:3])
-
-    official_total = 0
-    official_areas = official_buffer.area
-    for i in official_areas:
-        official_total += i
-    #print(official_areas[0:3])
-    print("osm areas: ", osm_total)
-    print("official areas: ", official_total)
-
-    intersected = gpd.clip(official_buffer, osm_buffer)
-
-    totalshare = 0
-    in_areas = intersected.area
-    for i in in_areas:
-        totalshare += i
-    print(totalshare)
-    percent_official_intersected = totalshare*100/official_total
-    print("intersected: ",totalshare)
-    print("intersected length: ", len(intersected))
-    print("percent_official_intersected: ", percent_official_intersected)
-    percent_dict[buff] = percent_official_intersected
-    percent_list.append((buff,percent_official_intersected))
-
-for item in percent_list:
-    print("buffer: ", item[0])
-    print("Percent area intersected: ", item[1])
-    print("-----------------------")
+
+# For 'filename', insert belfast.json, hong_kong.json, or olomouc.json
+def load_data(filename, city):
+    # Get the work directory
+    dirname = os.path.abspath('')
+    # Define the path for the city-specifc configuration file
+    jsonFile = 'configuration/' + filename
+    jsonPath = os.path.join(dirname, jsonFile)
+    # Use the defined paths to run the config file
+    try:
+        with open(jsonPath) as json_file:
+            config = json.load(json_file)
+    except Exception as e:
+        print('Failed to read json file.')
+        print(e)
+
+    # Read the filepaths so data can be accessed
+    graphml_path = os.path.join(dirname, 'data', city, config['graphml_path'])
+    osm_buffer_path = os.path.join(dirname, 'data', city, config['osm_buffer'])
+    gdf_official_path = os.path.join(dirname, 'data', city, config ['gdf_official'])
+
+    # Open the files from the filepaths
+    graphml = open(graphml_path)
+    osm_buffer = open(osm_buffer_path)
```

Comment:
@nicholas-david the problem is here. You're opening the file itself, then taking the open file and asking geopandas to open it. This should just return `osm_buffer_path`, that's all geopandas needs to open to the file.

Reply:
Thank you for clearing that up. The reason that I opened the file using those lines is that I was getting an error that says strings do not have the attribute .crs. when trying to change the crs to standardized them between data sources. 

Is there a way to fix that issue?

Reply:
I made that code edit that I suggested, then re-ran the code snippet you provided earlier, and it all seemed to run fine.